packages/cli/src/commands/export-dynamic-plugin/backend.ts

```diff
@@ -0,0 +1,320 @@
+  } = {};
+
+  const rollupConfig = rollupConfigs[0];
+  rollupConfig.plugins?.push(
+    embedModules({
+      filter: filter,
+      addDependency(embeddedModule, dependencyName, dependencyVersion) {
+        const existingVersion = dependenciesToAdd[dependencyName];
+        if (existingVersion === undefined) {
+          dependenciesToAdd[dependencyName] = dependencyVersion;
+          return;
+        }
+        if (existingVersion !== dependencyVersion) {
+          throw new Error(
+            `several versions ('${existingVersion}', '${dependencyVersion}') of the same transitive dependency ('${dependencyName}') for embedded module ('${embeddedModule}')`,
+          );
+        }
+      },
+    }),
+  );
+
+  if (Array.isArray(rollupConfig.output)) {
+    rollupConfig.output.forEach(output => {
+      if (output.format === 'commonjs') {
+        output.interop = 'default';
+      }
+    });
+  }
+
+  await fs.remove(paths.resolveTarget('dist'));
+
+  try {
+    const bundle = await rollup(rollupConfig);
+    if (rollupConfig.output) {
+      for (const output of [rollupConfig.output].flat()) {
+        await bundle.generate(output);
+        await bundle.write(output);
+      }
+    }
+  } catch (error) {
+    throw new Error(formatErrorMessage(error));
+  }
+
+  if (opts.clean) {
+    await fs.remove(target);
+  }
+
+  const monoRepoPackages = await getPackages(paths.targetDir);
+  await productionPack({
+    packageDir: '',
+    targetDir: target,
+    customizeManifest: (pkgToCustomize: BackstagePackageJson) => {
+      function test(str: string, expr: string | RegExp): boolean {
+        if (typeof expr === 'string') {
+          return str === expr;
+        }
+        return expr.test(str);
+      }
+
+      pkgToCustomize.name = `${pkgToCustomize.name}-dynamic`;
+      (pkgToCustomize as any).bundleDependencies = true;
+      pkgToCustomize.scripts = undefined;
+      pkgToCustomize.types = undefined;
+
+      pkgToCustomize.files = pkgToCustomize.files?.filter(
+        f => !f.startsWith('dist-dynamic/'),
+      );
+
+      for (const dep in dependenciesToAdd) {
+        if (!Object.prototype.hasOwnProperty.call(dependenciesToAdd, dep)) {
+          continue;
+        }
+        if (pkgToCustomize.dependencies === undefined) {
+          pkgToCustomize.dependencies = {};
+        }
+        const existingVersion = pkgToCustomize.dependencies[dep];
+        if (existingVersion === undefined) {
+          pkgToCustomize.dependencies[dep] = dependenciesToAdd[dep];
+          continue;
+        }
+        if (existingVersion !== dependenciesToAdd[dep]) {
+          if (!semver.satisfies(existingVersion, dependenciesToAdd[dep])) {
+            throw new Error(
+              `The version of a dependency ('${dep}') of an embedded module conflicts with main module dependencies: '${dependenciesToAdd[dep]}', '${existingVersion}'`,
+            );
+          }
+        }
+      }
+      if (pkgToCustomize.dependencies) {
```

Comment:
This is always true since the reset above assigns a `{}` here and `Boolean({}) === true`

Reply:
It is not (uness I'm misreading). It might be undefined if the `dependenciesToAdd` list was empty.